File: Lesson01_BlinkingLed/code/c/Lesson01_BlinkingLed.c
#include <lgpio.h>
#include <stdio.h>

#define LED_PIN 17  // BCM17 (physical pin 11)
int h;

int main(void)
{
    h = lgGpiochipOpen(4);
    if (h < 0) {
        printf("Failed to open gpiochip4, try gpiochip0\n");
        h = lgGpiochipOpen(0);
        if (h < 0) return -1;
    }

    lgGpioClaimOutput(h, 0, LED_PIN, 0);

    while (1) {
        lgGpioWrite(h, LED_PIN, 0);  // LED on (active low)
        printf("led on...\n");
        lguSleep(0.5);
        lgGpioWrite(h, LED_PIN, 1);  // LED off
        printf("...led off\n");
        lguSleep(0.5);
    }

    lgGpiochipClose(h);
    return 0;
}

File: Lesson01_BlinkingLed/code/python/Lesson01_BlinkingLed.py
from gpiozero import LED
import time

# BCM17 (physical pin 11), common anode LED (active low)
led = LED(17, active_high=False)

try:
    while True:
        print('...led on')
        led.on()
        time.sleep(0.5)
        print('led off...')
        led.off()
        time.sleep(0.5)
except KeyboardInterrupt:
    led.off()
